source/Halibut/ServiceModel/HalibutProxyWithAsync.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Halibut.Diagnostics;
+using Halibut.Exceptions;
+using Halibut.Transport.Protocol;
+
+namespace Halibut.ServiceModel
+{
+    public class HalibutProxyWithAsync : DispatchProxyAsync
+    {
+        Func<RequestMessage, MethodInfo, CancellationToken, Task<ResponseMessage>> messageRouter;
+        Type contractType;
+        ServiceEndPoint endPoint;
+        long callId;
+        bool configured;
+        CancellationToken globalCancellationToken;
+        ILog logger;
+
+        public void Configure(Func<RequestMessage, MethodInfo, CancellationToken, Task<ResponseMessage>> messageRouter, Type contractType, ServiceEndPoint endPoint, ILog logger, CancellationToken cancellationToken)
+        {
+            this.messageRouter = messageRouter;
+            this.contractType = contractType;
+            this.endPoint = endPoint;
+            this.globalCancellationToken = cancellationToken;
+            this.configured = true;
+            this.logger = logger;
+        }
+
+        public override object Invoke(MethodInfo targetMethod, object[] args)
+        {
+            throw new NotImplementedException();
```

Comment:
nit: NotSupportedException or something along those lines with a helpful message to the caller

Reply:
Good idea! TIL